query.go

```diff
@@ -364,18 +367,23 @@ func (q *query) queryPeer(ctx context.Context, ch chan<- *queryUpdate, p peer.ID
 	dialCtx, queryCtx := ctx, ctx
 
 	// dial the peer
-	if err := q.dht.dialPeer(dialCtx, p); err != nil {
+	if err := q.dht.dialPeer(dialCtx, p); err != nil && dialCtx.Err() == nil {
+		q.dht.peerStoppedDHT(q.dht.ctx, p)
```

Comment:
Did you mean to put the `dialCtx.Err() == nil` check only around the `q.dht.peerStoppedDHT`? We still need to return (setting the peer to unreachable is optional I think).

Reply:
Good catch.